### ISSUE TITLE
Cannot send 2 minute iPhone video

### ISSUE THREAD

**Issue by mailinglists35** [signalapp/Signal-iOS#5839]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
Attachment limit size is not documented. Trying to send a two minute iPhone video is refused with Attachment is too large. Too large than what? 

### Steps to reproduce
- Open iPhone Camera in default settings
- Film a two minute clip
- Tap the share button
- Pick Signal from the list

**Actual result:**
Attachment too large

**Expected result:** 
Send the clip

### Screenshots

Will upload later, submitting on mobile

### Device info

**Device**: iPhone 13 mini

**iOS version**: 17.0

**Signal version:** latest from App Store

**Comment by mailinglists35** [comment]
![image](https://github.com/user-attachments/assets/f09d716b-7f17-44a4-8da7-540a494de6a2)

**Comment by mailinglists35** [comment]
![image](https://github.com/user-attachments/assets/c643076b-c304-46dd-92f4-1127b1948111)

video exif

**Comment by mailinglists35** [comment]
error message

![image](https://github.com/user-attachments/assets/a508a8ec-173e-429e-b3ba-63b7676edf4b)

**Comment by mailinglists35** [comment]
![image](https://github.com/user-attachments/assets/51c0322e-7ecb-40ee-ab56-279d545da0c9)

**Comment by mailinglists35** [comment]
other apps accept it. facebook for example. Session app. Whatsapp. Messenger. Mail. virtually any other communication app than Signal

![image](https://github.com/user-attachments/assets/bd9af479-de92-4423-a64f-adc1188808e5)

**Comment by 0x1eaf** [comment]
Same here, was regularly sending videos of this length before, but now it stopped working.

**Comment by meteficha** [comment]
Also affected, but I’ll add that I can share the same video through the app itself just fine. The issue occurs only when sharing from the Photos gallery. 

It also only seems to happens if the videos were shot with “Most compatible” H.264 encoding (ironically). Switching to “High efficiency” HEVC encoding seems to sidestep this issue.

**Comment by mailinglists35** [comment]
yes, I only accidentally recently discovered it occurs only when sharing via ios share sheet, and not from within the app UI itself

**Comment by Jamo33** [comment]
I also have this issue

**Comment by ryancabanas** [comment]
I have this issue too. I even have my videos set on “High Efficiency”.  I’ve tried attaching it every which way and it still doesn’t work. Says my 1 minute video is too large.

**Comment by aleksfadini** [comment]
I abandoned signal for this reason. I need to send a longer videos.

**Comment by mrkobel** [comment]
I've recently switched from Android to iPhone and hit this issue. I was able to share full HD videos up to 2 minute long on Android Signal app. 

The limit seems to be much lower on iPhone (and unknown). The issue is well described by submitter. It's especially frustrating when I select 20+ photos and videos for sharing in iPhone gallery and get this error, then I need to start again. In fact i no longer share videos because of this bug - it makes it unpredictable. 

Please fix :)

**Comment by mailinglists35** [comment]
not even a 40 second fhd 60fps hdr video

![image](https://github.com/user-attachments/assets/b35ead2e-2d15-45b0-9173-37caea9aba2e)

**Comment by brandonnodnarb** [comment]
Just a quick chirp --- I (still) am experiencing this issue.

**Comment by Trufax** [comment]
I also face this issue. I have the feeling its worse with latest version. I basically can only send very short clips.

**Comment by akirataguchi115** [comment]
I think this is the real issue here:
> Attachment limit size is not documented

If the messages and files are stored on devices instead of servers, why limit the file size? Also why not document the file size limit somewhere?

**Comment by MarlowBrown** [comment]
Hello @mailinglists35 @0x1eaf @meteficha @Jamo33 @ryancabanas @aleksfadini @mrkobel @brandonnodnarb @Trufax and @akirataguchi115.

I am a contributor to the iOS repository! I want to make a patch for this, but I haven't been able to reproduce this bug on my end. Could you send me your videos that are triggering the bug here on Git Hub?

If you find a good video, make sure to remove any identifiable metadata (like GPS data) associated with it!

**Comment by mailinglists35** [comment]
@MarlowBrown I think it is easy reproducible by just following the steps described.

All you need is an iOS device and a two minute HDR HEIC video (which I believe is the default video type produced by modern iPhones)

**Comment by NeroVanbiervliet** [comment]
Started experiencing the same issue on iOS in a group chat about a month ago - multiple people running into it.

**Comment by RichardFevrier** [comment]
Please reopen

**Comment by mailinglists35** [comment]
this issue still persists, please reopen and fix it

**Comment by mailinglists35** [comment]
and for larger videos like 3 minutes it also fails even when sending from within the app.

**Comment by max-signal** [comment]
Update: The limit for videos is now around 200MiB. This should hopefully allow more videos to be sent than was possible before.

Documenting this based on the file size is difficult because it’s the post-transcoding size that’s enforced, and a given video could get larger (for compatibility reasons) or smaller (such as 4k becoming 1080p). We are continuing to make changes to the video sending flow, and I’ll circle back here when there’s better documentation/UX for which videos can/can’t be sent.

**Comment by rogersentongo** [comment]
Adding a controlled test that isolates the variable, since the thread so far has reports but not a
repro that separates size from duration.

**The rejection tracks duration, not file size.** I encoded the same 11:35 source film at several
sizes and durations with identical settings and tried to send each one:

| duration | size | result |
|---|---|---|
| 0:15 | 0.73 MB | sends |
| 1:00 | 9.97 MB | sends |
| 3:00 | 15.66 MB | sends |
| 5:48 | 24.60 MB | sends |
| 6:00 | 20.38 MB | sends |
| 11:35 | 5.98 MB | **"attachment too large"** |
| 11:35 | 15.64 MB | **"attachment too large"** |
| 11:35 | 29.56 MB | **"attachment too large"** |

A **5.98 MB** file is refused as "too large" while a **24.60 MB** file of the same video sends
without complaint. Size varies by more than 4× among the rejected/accepted pairs and does not
predict the outcome; duration does. The threshold sits somewhere between 6 minutes and 11:35.

This also explains the "it used to work" reports: shorter versions of the same material sent fine,
and re-encoding smaller — the obvious response to the error message — never helps, because size is
not the constraint being violated.

**Encoding details.** All files were H.264 Main@3.1, `yuv420p`, closed GOP, AAC-LC 44.1 kHz stereo
128 kbps, MP4 with `mp42` brand and `+faststart` (moov before mdat). SDR, no HDR/HEVC involved.
Verified with ffprobe: two streams only, no rotation matrix, no edit lists, decodes end-to-end with
no errors. Behaviour is identical whether attached from the photo picker or the Files/document
picker, and identical whether the file arrived by AirDrop or was already on the device.

**Likely mechanism.** If the client estimates the size of its own transcode as
`duration × target bitrate` and compares that against the attachment cap before encoding, then an
11:35 video at a ~2 Mbps target estimates to roughly 170 MB and is refused regardless of how small
the source actually is — which matches every observation above.

**Suggested fix**, in order of preference:

1. Report the real constraint — "This video is too long to send (maximum ~N minutes)" — so the
   user isn't sent to re-encode smaller, which cannot work.
2. Transcode to fit the cap rather than rejecting, since the source here is already far below it.
3. At minimum, document the effective duration limit.

**Environment**
- Signal iOS 8.24
- iOS 26.6
- iPhone 16e

Happy to provide the sample files if they'd help reproduce.
